app/ai_backend/test_genesis_evolutionary_conduit.py

```
@@ -0,0 +1,776 @@
+            self.conduit.get_best_agent([])
+            
+    def test_save_evolution_state_success(self):
+        """Test successful evolution state saving."""
+        self.temp_dir = tempfile.mkdtemp()
+        state_file = os.path.join(self.temp_dir, 'evolution_state.json')
+        
+        mock_population = [Mock() for _ in range(5)]
+        generation = 10
+        
+        with patch('builtins.open', mock_open()) as mock_file:
+            with patch('json.dump') as mock_json:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)